frontends/mit-open/src/pages/DepartmentListingPage/DepartmentListingPage.test.tsx

```diff
@@ -0,0 +1,160 @@
+import React from "react"
+import { renderWithProviders, screen, waitFor, within } from "@/test-utils"
+import type { LearningResourceSearchResponse } from "api"
+import DepartmentListingPage from "./DepartmentListingPage"
+import { factories, setMockResponse, urls } from "api/test-utils"
+import invariant from "tiny-invariant"
+import { faker } from "@faker-js/faker/locale/en"
+
+const makeSearchResponse = (
+  aggregations: Record<string, number>,
+): LearningResourceSearchResponse => {
+  return {
+    metadata: {
+      suggestions: [],
+      aggregations: {
+        department: Object.entries(aggregations).map(([key, docCount]) => ({
+          key,
+          doc_count: docCount,
+        })),
+      },
+    },
+    count: 0,
+    results: [],
+    next: null,
+    previous: null,
+  }
+}
+
+describe("DepartmentListingPage", () => {
+  const setupApis = () => {
```

Comment:
Do we need to setup APIs and render in each test? Seems we can run once in the `describe` block.

Reply:
It does need to be done per-test: `@testing-library/react` unmounts whatever it mounted after each test.

It could be done with a `beforeEach`, but I generally prefer to just call the setup function, which allows for easier per-test customizations and provides access to the generated data without `let ... = ` reassignments.